os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -174,7 +174,7 @@ int binary_manager_check_kernel_update(void)
 	snprintf(filepath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, kernel_info.part_info[inactive_partidx].devnum);
 	ret = binary_manager_read_header(BINARY_KERNEL, filepath, (void *)&header_data, true);
 	if (ret == BINMGR_OK) {
-		if (kernel_info.version < header_data.version) {
+		if (kernel_info.version <= header_data.version) {
```

Comment:
How about making config for same version update?
Because it can be different for policy.